caffe2/operators/cross_entropy_op.cc

```diff
@@ -78,13 +78,15 @@ bool SigmoidCrossEntropyWithLogitsOp<float, CPUContext>::RunOnDevice() {
   const auto inner_size = logits.dim() > 0 ? logits.sizes().back() : 1;
   const auto outer_size = logits.numel() / inner_size;
 
-  auto* out = Output(0);
+  std::vector<int64_t> dims;
   if (logits.dim() == 0) {
-    out->Resize(std::vector<int64_t>{});
+    dims = std::vector<int64_t>{};
   } else {
-    std::vector<int64_t> dims(logits.sizes().begin(), logits.sizes().end() - 1);
-    out->Resize(dims);
+    std::vector<int64_t> dims_(
+        logits.sizes().begin(), logits.sizes().end() - 1);
+    dims = dims_;
```

Comment:
This code seems overly complicated. Proposing some changes that make it more readable, and also faster (because it avoids the vector copy you have in line 87 and moves instead).

    std::vector<int64_t> dims;
    if (logits.dim() != 0) {
      dims = std::vector<int64_t>(
          logits.sizes().begin(), logits.sizes().end() - 1);
    }